frontend/src/test/scala/bloop/bsp/BspIntellijClientSpec.scala

```diff
@@ -0,0 +1,63 @@
+package bloop.bsp
+
+import java.io.File
+import java.net.URLClassLoader
+import java.nio.file.Files
+import java.nio.file.StandardOpenOption
+
+import bloop.cli.BspProtocol
+import bloop.data.WorkspaceSettings
+import bloop.io.AbsolutePath
+import bloop.logging.BspClientLogger
+import bloop.logging.RecordingLogger
+import bloop.util.{TestProject, TestUtil}
+import ch.epfl.scala.bsp.WorkspaceBuildTargetsResult
+
+object LocalBspIntellijClientSpec extends BspIntellijClientSpec(BspProtocol.Local)
+object TcpBspIntellijClientSpec extends BspIntellijClientSpec(BspProtocol.Tcp)
+
+class BspIntellijClientSpec(
+    override val protocol: BspProtocol
+) extends BspBaseSuite {
+
+  test("refresh project data on buildTargets request") {
+    TestUtil.withinWorkspace { workspace =>
+      val logger = new RecordingLogger(ansiCodesSupported = false)
+      val `A` = TestProject(workspace, "a", Nil)
+      val `B` = TestProject(workspace, "b", Nil)
+      val initialProjects = List(`A`)
+      val configDir = TestProject.populateWorkspace(workspace, initialProjects)
+
+      val refreshProjectsCommand = {
+        val content = sanitizeForBash(`B`.toJson)
+        val path = configDir.resolve("b.json")
+        val addProjectBCommand = s"echo $content > $path"
+        // as settings are read when project is initialized, with this command
+        // we simulate that project 'b' is added after first request
+        List(
+          "bash",
+          "-c",
+          s"[ ! -f first-run-marker ] && touch first-run-marker || $addProjectBCommand"
+        )
+      }
+
+      val workspaceSettings =
+        WorkspaceSettings(None, None, refreshProjectsCommand = Some(refreshProjectsCommand))
+      WorkspaceSettings.writeToFile(configDir, workspaceSettings, logger)
+
+      loadBspState(workspace, initialProjects, logger, bspClientName = "IntelliJ") { state =>
+        def projectNames(w: WorkspaceBuildTargetsResult) = w.targets.flatMap(_.displayName)
+
+        val initialTargets = state.workspaceTargets
+        assertEquals(projectNames(initialTargets), List("a"))
+        val finalTargets = state.workspaceTargets
+        assertEquals(projectNames(finalTargets), List("a", "b"))
```

Comment:
Amazing! Very nice to have this assertion 👍

Reply:
The Windows CI failure looks legitimate 

```diff
X bloop.bsp.TcpBspIntellijClientSpec.refresh project data on buildTargets request 274ms 
  bloop.testing.DiffAssertions$TestFailedException: failed assertion at D:\a\bloop\bloop\frontend\src\test\scala\bloop\testing\BaseSuite.scala:82
failed assertion at D:\a\bloop\bloop\frontend\src\test\scala\bloop\testing\DiffAssertions.scala:82
===========
=> Obtained
===========
List(a)

=======
=> Diff
=======
@@ -1,1 +1,1 @@
-List(a)
+List(a, b)
```

I wonder if the bash command is working correctly on Windows? It might be worth include the BSP logging statements to the assertion error.